File: app_simplified.js
// Global state
let currentImageData = null;
let generatedImages = [];

// DOM Elements
const ideaInput = document.getElementById('ideaInput');
const generateBtn = document.getElementById('generateBtn');
const imageGallery = document.getElementById('imageGallery');
const statusMessage = document.getElementById('statusMessage');
const modal = document.getElementById('imageModal');
const modalImage = document.getElementById('modalImage');
const modalPrompt = document.getElementById('modalPrompt');
const closeBtn = document.querySelector('.close-btn');
const download3DBtn = document.getElementById('download3DBtn');
const downloadImageBtn = document.getElementById('downloadImageBtn');
const modelStatus = document.getElementById('modelStatus');

// API Configuration
const API_BASE_URL = 'http://localhost:5000/api';

// Event Listeners
generateBtn.addEventListener('click', generateImage);
closeBtn.addEventListener('click', closeModal);
download3DBtn.addEventListener('click', convertTo3D);
downloadImageBtn.addEventListener('click', downloadImage);

// Close modal when clicking outside
window.addEventListener('click', (e) => {
    if (e.target === modal) {
        closeModal();
    }
});

// Allow Enter key to generate (with Shift+Enter for new line)
ideaInput.addEventListener('keydown', (e) => {
    if (e.key === 'Enter' && !e.shiftKey) {
        e.preventDefault();
        generateImage();
    }
});

/**
 * Main function to generate image from user idea
 */
async function generateImage() {
    const idea = ideaInput.value.trim();
    
    if (!idea) {
        showStatus('Please enter your idea or inspiration!', 'error');
        return;
    }
    
    // Disable button and show loading state
    setLoadingState(true);
    showStatus('🎨 Generating your image with DALL-E 3... This may take 10-15 seconds.', 'info');
    
    try {
        // Call Flask backend
        const response = await fetch(`${API_BASE_URL}/generate-image`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({ idea: idea })
        });
        
        if (!response.ok) {
            const errorData = await response.json();
            throw new Error(errorData.error || `HTTP error! status: ${response.status}`);
        }
        
        const data = await response.json();
        
        if (data.success) {
            // Add image to gallery
            const imageData = {
                id: data.image_id,
                imageUrl: `http://localhost:5000${data.image_url}`,
                imagePath: data.image_path,  // Store for 3D conversion
                prompt: data.prompt,
                timestamp: new Date().toISOString()
            };
            
            addImageToGallery(imageData);
            
            showStatus('✨ Image generated successfully! Click to view or convert to 3D.', 'success');
            
            // Clear input
            ideaInput.value = '';
            
        } else {
            throw new Error(data.error || 'Failed to generate image');
        }
        
    } catch (error) {
        console.error('Error generating image:', error);
        showStatus(`❌ Error: ${error.message}`, 'error');
    } finally {
        setLoadingState(false);
    }
}

/**
 * Add generated image to the gallery
 */
function addImageToGallery(imageData) {
    generatedImages.unshift(imageData);  // Add to beginning
    
    const galleryItem = document.createElement('div');
    galleryItem.className = 'gallery-item';
    galleryItem.dataset.imageId = imageData.id;
    
    galleryItem.innerHTML = `
        <img src="${imageData.imageUrl}" alt="${imageData.prompt}" loading="lazy">
        <div class="gallery-item-overlay">
            <p class="gallery-item-prompt">${imageData.prompt}</p>
        </div>
    `;
    
    galleryItem.addEventListener('click', () => openModal(imageData));
    
    // Insert at beginning of gallery
    imageGallery.insertBefore(galleryItem, imageGallery.firstChild);
    
    // Add entrance animation
    setTimeout(() => {
        galleryItem.style.animation = 'fadeIn 0.5s';
    }, 10);
}

/**
 * Open modal to view image details
 */
function openModal(imageData) {
    currentImageData = imageData;
    modalImage.src = imageData.imageUrl;
    modalPrompt.textContent = imageData.prompt;
    modal.style.display = 'block';
    modelStatus.style.display = 'none';
    modelStatus.className = 'model-status';
    download3DBtn.disabled = false;
}

/**
 * Close the modal
 */
function closeModal() {
    modal.style.display = 'none';
    currentImageData = null;
}

/**
 * Convert current image to 3D model using Meshy AI
 */
async function convertTo3D() {
    if (!currentImageData) return;
    
    modelStatus.className = 'model-status loading';
    modelStatus.textContent = '🎭 Converting to 3D model... This may take 2-3 minutes. Please wait.';
    download3DBtn.disabled = true;
    
    try {
        const response = await fetch(`${API_BASE_URL}/convert-to-3d`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                image_path: currentImageData.imagePath
            })
        });
        
        if (!response.ok) {
            const errorData = await response.json();
            throw new Error(errorData.error || `HTTP error! status: ${response.status}`);
        }
        
        const data = await response.json();
        
        if (data.success) {
            const modelUrl = `http://localhost:5000${data.model_url}`;
            modelStatus.className = 'model-status success';
            modelStatus.innerHTML = `
                ✅ 3D Model ready! 
                <br><br>
                <a href="${modelUrl}" download class="download-link">
                    📥 Download 3D Model (.glb)
                </a>
                <br><br>
                <small>You can view .glb files in Windows 3D Viewer, Blender, or online at <a href="https://gltf-viewer.donmccurdy.com/" target="_blank">glTF Viewer</a></small>
            `;
        } else {
            throw new Error(data.error || 'Failed to convert to 3D');
        }
        
    } catch (error) {
        console.error('Error converting to 3D:', error);
        modelStatus.className = 'model-status error';
        modelStatus.textContent = `❌ Error: ${error.message}`;
    } finally {
        download3DBtn.disabled = false;
    }
}

/**
 * Download the current image
 */
function downloadImage() {
    if (!currentImageData) return;
    
    // Create a temporary link and trigger download
    const link = document.createElement('a');
    link.href = currentImageData.imageUrl;
    link.download = `ai-generated-${currentImageData.id}.png`;
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
    
    showStatus('📥 Image downloaded!', 'success');
}

/**
 * Show status message to user
 */
function showStatus(message, type = 'info') {
    statusMessage.textContent = message;
    statusMessage.className = `status-message ${type}`;
    
    // Auto-hide success messages after 5 seconds
    if (type === 'success') {
        setTimeout(() => {
            if (statusMessage.className.includes('success')) {
                statusMessage.style.display = 'none';
            }
        }, 5000);
    }
}

/**
 * Set loading state for generate button
 */
function setLoadingState(isLoading) {
    generateBtn.disabled = isLoading;
    const btnText = generateBtn.querySelector('.btn-text');
    const btnLoader = generateBtn.querySelector('.btn-loader');
    
    if (isLoading) {
        btnText.style.display = 'none';
        btnLoader.style.display = 'inline';
    } else {
        btnText.style.display = 'inline';
        btnLoader.style.display = 'none';
    }
}

// Test API connection on load
window.addEventListener('load', async () => {
    try {
        const response = await fetch(`${API_BASE_URL}/health`);
        const data = await response.json();
        console.log('✅ Backend connected:', data);
    } catch (error) {
        console.error('❌ Backend connection failed:', error);
        showStatus('⚠️ Backend not connected. Make sure Flask server is running on port 5000.', 'error');
    }
});

File: app_simplified.py
"""
Simplified Flask Backend for AI Image Generator
- Direct OpenAI DALL-E 3 image generation
- Meshy AI 3D model conversion
"""

from flask import Flask, request, jsonify, send_from_directory
from flask_cors import CORS
import os
import requests
import uuid
from datetime import datetime
import time
import json

app = Flask(__name__)
CORS(app)

# ==================== CONFIGURATION ====================
# Your API Keys
OPENAI_API_KEY = ""
MESHY_API_KEY = ""

# Storage folders
UPLOAD_FOLDER = 'static/generated_images'
MODELS_FOLDER = 'static/3d_models'
os.makedirs(UPLOAD_FOLDER, exist_ok=True)
os.makedirs(MODELS_FOLDER, exist_ok=True)

# ==================== HELPER FUNCTIONS ====================

def generate_image_dalle(prompt):
    """
    Generate image using OpenAI DALL-E 3
    """
    try:
        print(f"📸 Generating image with DALL-E 3...")
        print(f"Prompt: {prompt}")
        
        url = "https://api.openai.com/v1/images/generations"
        
        headers = {
            "Authorization": f"Bearer {OPENAI_API_KEY}",
            "Content-Type": "application/json"
        }
        
        payload = {
            "model": "dall-e-3",
            "prompt": prompt,
            "n": 1,
            "size": "1024x1024",
            "quality": "standard"
        }
        
        response = requests.post(url, json=payload, headers=headers, timeout=60)
        
        if response.status_code != 200:
            error_msg = response.json().get('error', {}).get('message', 'Unknown error')
            raise Exception(f"OpenAI API error: {error_msg}")
        
        data = response.json()
        image_url = data['data'][0]['url']
        
        print(f"✅ Image generated: {image_url}")
        
        # Download and save image locally
        image_response = requests.get(image_url, timeout=30)
        image_filename = f"{uuid.uuid4()}.png"
        image_path = os.path.join(UPLOAD_FOLDER, image_filename)
        
        with open(image_path, 'wb') as f:
            f.write(image_response.content)
        
        local_url = f"/static/generated_images/{image_filename}"
        print(f"💾 Image saved locally: {local_url}")
        
        return local_url, image_path
        
    except Exception as e:
        print(f"❌ Error generating image: {e}")
        raise Exception(f"Image generation failed: {str(e)}")


def convert_to_3d_meshy(image_path):
    """
    Convert image to 3D model using Meshy AI
    
    Meshy AI Workflow:
    1. Upload image and create task
    2. Poll task status until complete
    3. Download 3D model
    """
    try:
        print(f"🎭 Starting 3D conversion with Meshy AI...")
        
        # Step 1: Create image-to-3D task
        url = "https://api.meshy.ai/v2/image-to-3d"
        
        headers = {
            "Authorization": f"Bearer {MESHY_API_KEY}"
        }
        
        # Read image file
        with open(image_path, 'rb') as f:
            image_data = f.read()
        
        # Create task
        files = {
            'image_file': ('image.png', image_data, 'image/png')
        }
        
        data = {
            'enable_pbr': 'true',  # Enable PBR materials
            'ai_model': 'meshy-4'  # Latest model
        }
        
        print("📤 Uploading image to Meshy AI...")
        response = requests.post(url, headers=headers, files=files, data=data, timeout=30)
        
        if response.status_code not in [200, 201]:
            error_msg = response.text
            raise Exception(f"Meshy API error: {error_msg}")
        
        result = response.json()
        task_id = result.get('result')
        
        if not task_id:
            raise Exception("No task ID returned from Meshy")
        
        print(f"✅ Task created: {task_id}")
        print("⏳ Waiting for 3D model generation (this may take 2-3 minutes)...")
        
        # Step 2: Poll for task completion
        status_url = f"https://api.meshy.ai/v2/image-to-3d/{task_id}"
        
        max_attempts = 60  # 5 minutes max (5 seconds per attempt)
        attempt = 0
        
        while attempt < max_attempts:
            time.sleep(5)  # Wait 5 seconds between checks
            attempt += 1
            
            status_response = requests.get(status_url, headers=headers, timeout=30)
            
            if status_response.status_code != 200:
                print(f"⚠️ Status check failed: {status_response.text}")
                continue
            
            status_data = status_response.json()
            status = status_data.get('status')
            progress = status_data.get('progress', 0)
            
            print(f"📊 Status: {status} | Progress: {progress}%")
            
            if status == 'SUCCEEDED':
                # Get model download URL
                model_urls = status_data.get('model_urls', {})
                glb_url = model_urls.get('glb')
                
                if not glb_url:
                    raise Exception("No GLB model URL in response")
                
                print(f"✅ 3D model ready! Downloading...")
                
                # Download 3D model
                model_response = requests.get(glb_url, timeout=60)
                model_filename = f"{uuid.uuid4()}.glb"
                model_path = os.path.join(MODELS_FOLDER, model_filename)
                
                with open(model_path, 'wb') as f:
                    f.write(model_response.content)
                
                local_model_url = f"/static/3d_models/{model_filename}"
                print(f"💾 3D model saved: {local_model_url}")
                
                return local_model_url
                
            elif status == 'FAILED':
                error = status_data.get('error', 'Unknown error')
                raise Exception(f"3D conversion failed: {error}")
            
            # Still processing, continue polling
        
        raise Exception("3D conversion timed out after 5 minutes")
        
    except Exception as e:
        print(f"❌ 3D conversion error: {e}")
        raise Exception(f"3D conversion failed: {str(e)}")


# ==================== API ROUTES ====================

@app.route('/api/generate-image', methods=['POST'])
def generate_image():
    """
    Generate image from user prompt (no Claude enhancement)
    """
    try:
        data = request.get_json()
        user_prompt = data.get('idea', '').strip()
        
        if not user_prompt:
            return jsonify({
                'success': False,
                'error': 'No prompt provided'
            }), 400
        
        print(f"\n{'='*60}")
        print(f"🎨 New image generation request")
        print(f"User prompt: {user_prompt}")
        print(f"{'='*60}\n")
        
        # Generate image directly with user's prompt
        image_url, image_path = generate_image_dalle(user_prompt)
        
        # Generate unique ID
        image_id = str(uuid.uuid4())
        
        print(f"\n✨ Image generation complete!")
        print(f"Image ID: {image_id}")
        print(f"Image URL: {image_url}\n")
        
        return jsonify({
            'success': True,
            'image_id': image_id,
            'image_url': image_url,
            'image_path': image_path,  # Store for 3D conversion
            'prompt': user_prompt,
            'timestamp': datetime.now().isoformat()
        })
        
    except Exception as e:
        print(f"\n❌ Error: {e}\n")
        return jsonify({
            'success': False,
            'error': str(e)
        }), 500


@app.route('/api/convert-to-3d', methods=['POST'])
def convert_to_3d():
    """
    Convert generated image to 3D model using Meshy AI
    """
    try:
        data = request.get_json()
        image_path = data.get('image_path')
        
        if not image_path:
            return jsonify({
                'success': False,
                'error': 'No image path provided'
            }), 400
        
        # Convert relative URL to file path if needed
        if image_path.startswith('/static/'):
            image_path = image_path.replace('/static/', 'static/')
        
        print(f"\n{'='*60}")
        print(f"🎭 3D conversion request")
        print(f"Image path: {image_path}")
        print(f"{'='*60}\n")
        
        # Convert to 3D
        model_url = convert_to_3d_meshy(image_path)
        
        print(f"\n✨ 3D conversion complete!")
        print(f"Model URL: {model_url}\n")
        
        return jsonify({
            'success': True,
            'model_url': model_url
        })
        
    except Exception as e:
        print(f"\n❌ 3D conversion error: {e}\n")
        return jsonify({
            'success': False,
            'error': str(e)
        }), 500


@app.route('/api/health', methods=['GET'])
def health_check():
    """Health check endpoint"""
    return jsonify({
        'status': 'healthy',
        'openai_configured': bool(OPENAI_API_KEY),
        'meshy_configured': bool(MESHY_API_KEY),
        'timestamp': datetime.now().isoformat()
    })


# Serve static files
@app.route('/static/<path:path>')
def serve_static(path):
    return send_from_directory('static', path)


# ==================== MAIN ====================

if __name__ == '__main__':
    print("\n" + "="*60)
    print("🚀 AI Image Generator Backend Starting...")
    print("="*60)
    print(f"📸 OpenAI DALL-E: {'✅ Configured' if OPENAI_API_KEY else '❌ Not configured'}")
    print(f"🎭 Meshy AI: {'✅ Configured' if MESHY_API_KEY else '❌ Not configured'}")
    print(f"💾 Upload folder: {UPLOAD_FOLDER}")
    print(f"🎨 Models folder: {MODELS_FOLDER}")
    print("="*60)
    print("\n🌐 Server running on http://localhost:5000")
    print("📡 API endpoints:")
    print("   POST /api/generate-image")
    print("   POST /api/convert-to-3d")
    print("   GET  /api/health")
    print("\n" + "="*60 + "\n")
    
    app.run(debug=True, host='0.0.0.0', port=5000)
